### PR TITLE
[FIXED JENKINS-33681] Plugin filters were failing to be removed and blocking restart

### PR DESCRIPTION
**Pull request by stephenc**
See [JENKINS-33681](https://issues.jenkins-ci.org/browse/JENKINS-33681)

@reviewbybees @jenkinsci/code-reviewers

### CODE REVIEW

**Comment by jtnord** [comment]
:bee:

File: core/src/main/java/hudson/util/PluginServletFilter.java [hunk 1]
```diff
@@ -148,18 +149,26 @@ public void destroy() {
     public static void cleanUp() {
         PluginServletFilter instance = getInstance(Jenkins.getInstance().servletContext);
         if (instance != null) {
-            for (Iterator<Filter> iterator = instance.list.iterator(); iterator.hasNext(); ) {
-                Filter f = iterator.next();
+            // While we could rely on the current implementation of list being a CopyOnWriteArrayList
+            // safer to just take an explicit copy of the list and operate on the copy
+            for (Filter f: new ArrayList<>(instance.list)) {
+                instance.list.remove(f);
+                // remove from the list even if destroy() fails as a failed destroy is still a destroy
```
Comment (varmenise) [inline core/src/main/java/hudson/util/PluginServletFilter.java]: this comment should probably be one line up

**Comment by varmenise** [comment]
:bee:

**Comment by stephenc** [comment]
Failures are all this random failure mode:

```
WARNING: Failed to scout com.cloudbees.plugins.credentials.SystemCredentialsProvider$ProviderImpl
java.lang.NoClassDefFoundError: Could not initialize class com.cloudbees.plugins.credentials.SystemCredentialsProvider$ProviderImpl
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:274)
	at hudson.ExtensionFinder$Sezpoz.scout(ExtensionFinder.java:675)
	at hudson.ClassicPluginStrategy.findComponents(ClassicPluginStrategy.java:383)
	at hudson.ExtensionList.load(ExtensionList.java:349)
	at hudson.ExtensionList.ensureLoaded(ExtensionList.java:287)
	at hudson.ExtensionList.iterator(ExtensionList.java:156)
	at hudson.ClassicPluginStrategy.findComponents(ClassicPluginStrategy.java:382)
	at hudson.ExtensionList.load(ExtensionList.java:349)
	at hudson.ExtensionList.ensureLoaded(ExtensionList.java:287)
	at hudson.ExtensionList.get(ExtensionList.java:171)
	at hudson.PluginManager$PluginUpdateMonitor.getInstance(PluginManager.java:1347)
	at hudson.maven.PluginImpl.init(PluginImpl.java:58)
	at sun.reflect.GeneratedMethodAccessor407.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at hudson.init.TaskMethodFinder.invoke(TaskMethodFinder.java:106)
	at hudson.init.TaskMethodFinder$TaskImpl.run(TaskMethodFinder.java:175)
	at org.jvnet.hudson.reactor.Reactor.runTask(Reactor.java:282)
	at jenkins.model.Jenkins$8.runTask(Jenkins.java:958)
	at org.jvnet.hudson.reactor.Reactor$2.run(Reactor.java:210)
	at org.jvnet.hudson.reactor.Reactor$Node.run(Reactor.java:117)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
```

**Pull request merged**